code/game/gamemodes/cm_initialize.dm

```diff
@@ -830,3 +831,82 @@ Additional game mode variables.
 
 /datum/game_mode/proc/get_escape_menu()
 	return "On the [SSmapping.configs[SHIP_MAP].map_name], orbiting..."
+
+
+//===================================================\\
+
+				//JOE INITIALIZE\\
+
+//===================================================\\
+
+/datum/game_mode/proc/initialize_joe(mob/living/carbon/human/joe)
+	joes += joe.ckey
+
+/datum/game_mode/proc/attempt_to_join_as_joe(mob/joe_candidate)
+	var/mob/living/carbon/human/new_joe = transform_joe(joe_candidate) //Initialized and ready.
+	if(!new_joe)
+		return
+
+	msg_admin_niche("Ghost ([new_joe.key]) has joined as Working Joe, [new_joe.real_name].")
+
+	if(joe_candidate)
+		joe_candidate.moveToNullspace() //Nullspace it for garbage collection later.
+
+/datum/game_mode/proc/check_joe_late_join(mob/joe_candidate, show_warning = 1)
+
+	if(!joe_candidate.client)
+		return
+
+	var/datum/job/joe_job = RoleAuthority.roles_by_name[JOB_WORKING_JOE]
+
+	if(!joe_job)
+		if(show_warning) to_chat(joe_candidate, SPAN_WARNING("Something went wrong!"))
+		return
+
+	if(!(RoleAuthority.roles_whitelist[joe_candidate.ckey] & WHITELIST_JOE))
+		if(show_warning) to_chat(joe_candidate, SPAN_WARNING("You are not whitelisted! You may apply on the forums to be whitelisted as a synth."))
```

Comment:
This seems to have reverted to the same line again. Please move the to_chat() to the next line.

Reply:
53ebddda58658b0f8c7e0bdfa913b7595d15426e